pretraining/utils/train_utils.py

```diff
@@ -44,18 +44,20 @@ def train(
         loss = ce_loss(output.view(-1, output.size(-1)), label.view(-1).long())
 
         loss.backward()
+        ddp_stats[1] += model.clip_grad_norm_(cfg.grad_clip_thresh).item()
```

Comment:
Does this function exist on module too? I think it's usually `clip_grad_norm_(model.parameters(), threshold)` ?

Also, this isn't just getting the norm but clipping the gradients in-place too right?The PR description makes me think this was intending to be a tracking-only change.

Reply:
Yeah it's usually an outside function, but because model is FSDP, FSDP has its own implementation (since clipping of sharded grads requires gradient communication). I added grad clipping because we should have been doing that to begin with - for monitoring only, the user can simply set the clipping threshold arbitrarily high (or we could just default it that way)

Reply:
I see... so we would set the threshold to something like float('inf') to reproduce how our original model was trained?

Reply:
Exactly, and that's true, I'll update the title/description. Let me take a look at the conflicts...